packages/components/src/components/select-combobox/src/index.tsx

```diff
@@ -11,50 +12,68 @@ import {
   StyledSelectElement as SelectComboboxElement,
   StyledSelectPopover as SelectComboboxPopover,
   StyledComboboxList as SelectComboboxList,
+  StyledSelectClearButton,
 } from "./styles";
 
 export interface SelectComoboxElementProps {
   comboboxState: ComboboxState;
   id: string;
   label: string;
   selectState: SelectState;
+  hasClearButton?: boolean;
 }
 
 const SelectCombobox = React.forwardRef<
   HTMLButtonElement,
   SelectComoboxElementProps
->(({ comboboxState, id, label, selectState, ...props }, ref) => {
-  return (
-    <>
-      <VisuallyHidden>
-        <Label htmlFor={id}>{label}</Label>
-      </VisuallyHidden>
-      <InputBox hasHover {...props}>
-        <SelectComboboxElement state={selectState} id={id} ref={ref} />
-      </InputBox>
-      <SelectComboboxPopover state={selectState} composite={false}>
-        <Box css={{ marginBottom: "$25" }}>
-          <InputBox>
-            <StyledCombobox
-              state={comboboxState}
-              autoSelect
-              placeholder="Search..."
-            />
-          </InputBox>
-        </Box>
-        <SelectComboboxList state={comboboxState}>
-          {comboboxState.matches.map((itemValue, i) => (
-            <StyledComboboxItem key={itemValue + i} focusOnHover>
-              {(itemProps) => (
-                <SelectComboboxItem {...itemProps} value={itemValue} />
-              )}
-            </StyledComboboxItem>
-          ))}
-        </SelectComboboxList>
-      </SelectComboboxPopover>
-    </>
-  );
-});
+>(
+  (
+    { comboboxState, id, hasClearButton, label, selectState, ...props },
+    ref
+  ) => {
+    return (
+      <>
+        <VisuallyHidden>
+          <Label htmlFor={id}>{label}</Label>
+        </VisuallyHidden>
+        <InputBox hasHover {...props}>
+          <SelectComboboxElement state={selectState} id={id} ref={ref} />
+          {hasClearButton &&
+            selectState.value !== ("" as SelectState["value"]) && (
+              <StyledSelectClearButton onClick={() => selectState.setValue("")}>
+                <CloseIcon
+                  decorative={false}
+                  title="Remove selection"
+                  color="current"
+                  size="xxsmall"
+                />
+              </StyledSelectClearButton>
+            )}
+        </InputBox>
+        <SelectComboboxPopover state={selectState} composite={false}>
+          <Box css={{ marginBottom: "$25" }}>
+            <InputBox>
+              <StyledCombobox
+                state={comboboxState}
+                autoSelect
+                placeholder="Search..."
+              />
+            </InputBox>
+          </Box>
+          <SelectComboboxList state={comboboxState}>
+            {comboboxState.matches.map((itemValue, i) => (
+              <StyledComboboxItem key={itemValue + i} focusOnHover>
```

Comment:
We should remove combining the itemValue with the index for the key. It's essentially no different than just using index for the key, which isn't great.
```suggestion
              <StyledComboboxItem key={itemValue} focusOnHover>
```